kitsune/sumo/middleware.py

```diff
@@ -70,6 +73,55 @@ def process_request(self, request):
             return super(SUMORefreshIDTokenAdminMiddleware, self).process_request(request)
 
 
+class ValidateAccessTokenMiddleware(SessionRefresh):
+    """Validate the ID Token from FxA every hour."""
+
+    def __init__(self, get_response=None):
+        if not settings.OIDC_ENABLE or settings.DEV:
+            raise MiddlewareNotUsed
+        super(ValidateAccessTokenMiddleware, self).__init__(get_response=get_response)
+
+    @staticmethod
+    def get_settings(attr, *args):
+        """Override settings for Firefox Accounts."""
+        val = get_oidc_fxa_setting(attr)
+        if val is not None:
+            return val
+        return super(ValidateAccessTokenMiddleware, ValidateAccessTokenMiddleware).get_settings(
+            attr, *args
+        )
+
+    def process_request(self, request):
+        """Check the validity of the id_token every hour"""
+        if not self.is_refreshable_url(request):
+            return
+
+        # the oidc_id_token_expiration key is set by the mozilla-django-oidc library
+        expiration = request.session.get("oidc_id_token_expiration", 0)
+        now = time.time()
+        access_token = request.session.get("oidc_access_token")
+
+        if expiration < now:
+
+            response_token_info = (
+                requests.post(settings.FXA_VERIFY_URL, data={"token": access_token})
+            ).json()
+
+            # if the token is not verified, log the user out
+            if (
+                response_token_info.get("code") == 400
+                and response_token_info.get("message") == "Invalid token"
+            ):
+                profile = request.user.profile
+                profile.fxa_refresh_token = ""
+                profile.save()
+                logout(request)
+            else:
+                request.session["oidc_id_token_expiration"] = (
+                    now + settings.FXA_RENEW_ID_TOKEN_EXPIRY_SECONDS
+                )
```

Comment:
We should add a comment here clarifying that, as you explained to me, we don't need to refresh the access_token since the verify endpoint will return fine, even for expired (not invalidated) tokens.